lib/codit/population/networks/city.py

```diff
@@ -148,3 +172,24 @@ def build_workplaces(people, classroom_size=-1):
 
 def next_workplace_size():
     return random.choice(household_workplace.WORKPLACE_SIZE_REPRESENTATIVE_EXAMPLES)
+
+
+def next_household_home(homes_list, allocated_coordinates_list):
+    """
+    Randomly pick up one home with ['lon', 'lat', 'building_type'], and remove the allocated home from the list,
+    as the number of homes are pre-defined according to actual number of accommodation of buildings and average number of
+    households per building_type, the actual number of households may exceed the pre-defined number of homes.
+    So here we recycle the apartments and terraces homes to allocate again just in case, as these two building types can
+    have multiple households.
+    :param homes_list:
+    :param allocated_coordinates_list:
+    :return: one home ['lon', 'lat', 'building_type']
+    """
+    if len(homes_list) > 0:
+        next_home = random.choice(homes_list)
```

Comment:
Is there any way to do choice without replacement at this point in the code? Possibly not ...

Reply:
I think I can simply use the full home coordinates list to generate random home address, as the number of households in apartments and terrace has been reflected in the full home list. The random home address picked would be generally correct, I don't have to worry too many households are allocated to detached houses or houses' coordinates, as those coordinates' occurrence rate is very low, only once in the home list.